### PR TITLE
Add quickstart introduction to repositories & archives (fixes #3620)

### LINKED ISSUE #3620: add some conceptual intro to quickstart docs
explain what a repository is, what an archive is.
otherwise it is hard to understand for people who never used git, tar, etc. before.

anything else?

### PR DESCRIPTION
**Pull request by milkey-mouse**
I took a shot at this description as well, sorry for kind of stealing this one @bhaveshpoddar94!

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -13,6 +13,17 @@ A step by step example
 
 .. include:: quickstart_example.rst.inc
 
+Archives and repositories
+-------------------------
+
```
Comment (ThomasWaldmann) [inline docs/quickstart.rst]: Guess we need to explain chunk first:

Chunk: a data (or metadata) block. To implement variable block size deduplication, borg cuts data (and metadata) streams into pieces. These pieces are stored into the repository and are addressed by their content hash.

File: docs/quickstart.rst [hunk 2]
```diff
@@ -13,6 +13,17 @@ A step by step example
...
+A Borg archive is used similarly to a tarball in a simpler backup system. When ``borg create``
+is run on a path, the current state of the files therein is stored in an archive. One can later
+extract or mount an archive to restore from a backup.
+
+Repositories are self-contained stores of archives that can be stored locally or remotely.
+Archives stored inside a repository are deduplicated with one another, which is why Borg
+shines for incremental backups.
+
```
[the remark this answers is not in the document]
Comment (milkey-mouse) [reply docs/quickstart.rst]: Do we really want the description to be that low level in the quickstart? How often will the end user care about chunks or the manifest?

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply docs/quickstart.rst]: Well, if we don't want to talk about chunks (or references to them), is it still clear/good enough?:

A borg archive is the named result of a backup run (borg create), referencing all data blocks for all backed up filesystem objects of that run.

A borg repository is a filesystem directory containing a storage for data blocks (usually created by backup runs), an index, locks and other related stuff.

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply docs/quickstart.rst]: ... a deduplicated storage ...

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply docs/quickstart.rst]: Is that ok? Can you update the PR?

**Comment by milkey-mouse** [comment]
OK, how's this? I tried to work in your more technical explanation while still making it helpful for a layman reading the quickstart for the first time.

**Comment by ThomasWaldmann** [comment]
OK, let's keep it like that for now. Thanks!

**Pull request merged**